distribution/xfer/download.go

```diff
@@ -35,9 +39,10 @@ func (ldm *LayerDownloadManager) SetConcurrency(concurrency int) {
 // NewLayerDownloadManager returns a new LayerDownloadManager.
 func NewLayerDownloadManager(layerStore layer.Store, concurrencyLimit int, options ...func(*LayerDownloadManager)) *LayerDownloadManager {
 	manager := LayerDownloadManager{
-		layerStore:   layerStore,
-		tm:           NewTransferManager(concurrencyLimit),
-		waitDuration: time.Second,
+		layerStore:      layerStore,
+		tm:              NewTransferManager(concurrencyLimit),
+		waitDuration:    time.Second,
+		cacheDirManager: cachedir.NewManager(filepath.Join(os.TempDir(), "resumable_downloads")),
```

Comment:
Just discovered while testing this that the daemon overrides `$TMPDIR` to point to `/var/lib/docker/tmp`. So this happens to put the files approximately where I wanted. But it's probably even better to explicitly put them somewhere under `config.Root`.